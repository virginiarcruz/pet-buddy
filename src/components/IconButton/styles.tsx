import styled, { css } from 'styled-components'
import ArrowIcon from '../Button/assets/arrow.svg'

export const IconButtonContainer = styled.div`
  ${({ theme }) => css`
    border-radius: ${theme.border.radius.pill};
    background-color: ${theme.color.blue};
    color: ${theme.color.white};
    width: 24px;
    height: 24px;
    cursor: pointer;

    display: flex;
    align-items: center;
    justify-content: center;
  `}
`

export const Icon = styled(ArrowIcon)`
  ${({ theme }) => css`
    width: 7px;
    height: 12px;
    transform: rotate(90deg);

    path {
      fill: ${theme.color.white};
    }
  `}
`
