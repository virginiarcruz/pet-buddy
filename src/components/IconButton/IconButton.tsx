import { IconButtonContainer, Icon } from './styles'

const IconButton = ({ ...props }) => (
  <IconButtonContainer className="icon-button" {...props}>
    <Icon />
  </IconButtonContainer>
)

export default IconButton
