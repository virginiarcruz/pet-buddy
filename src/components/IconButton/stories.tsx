import { Story, Meta } from '@storybook/react/types-6-0'
import IconButton from '.'

export default {
  title: 'IconButton',
  component: IconButton
} as Meta

export const Default: Story = () => <IconButton />
