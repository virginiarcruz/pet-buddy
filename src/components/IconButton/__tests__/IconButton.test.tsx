import { renderWithTheme } from 'utils/tests/helpers'

import IconButton from '../IconButton'

describe('<IconButton />', () => {
  it('should render the <IconButton />', () => {
    const { container } = renderWithTheme(<IconButton />)

    expect(container).toMatchSnapshot()
  })
})
