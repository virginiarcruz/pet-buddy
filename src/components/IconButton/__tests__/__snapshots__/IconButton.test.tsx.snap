// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> should render the <IconButton /> 1`] = `
.c0 {
  border-radius: 100px;
  background-color: #002FD6;
  color: #FFFFFF;
  width: 24px;
  height: 24px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 7px;
  height: 12px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c1 path {
  fill: #FFFFFF;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 7 12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.821 6.454a.665.665 0 000-.908L1.687.186c-.264-.248-.661-.248-.9 0L.18.818c-.239.247-.239.66 0 .935l4.075 4.233L.18 10.247c-.239.275-.239.688 0 .935l.608.632c.239.248.636.248.9 0l5.134-5.36z"
      />
    </svg>
  </div>
</div>
`;
